clients/runtime/src/rpc.rs

```diff
@@ -301,25 +342,24 @@ impl SpacewalkParachain {
 	async fn query_finalized_or_default<Address>(
 		&self,
 		address: Address,
-	) -> Result<<Address::Target as DecodeWithMetadata>::Target, Error>
+	) -> Result<Address::Target, Error>
 	where
 		Address: StorageAddress<IsFetchable = Yes, IsDefaultable = Yes>,
 	{
-		let hash = self.get_finalized_block_hash().await?;
-		Ok(self.api.storage().fetch_or_default(&address, hash).await?)
+		Ok(self.api.storage().at_latest().await.unwrap().fetch_or_default(&address).await?)
 	}
 
 	pub async fn get_finalized_block_hash(&self) -> Result<Option<H256>, Error> {
-		Ok(Some(self.api.rpc().finalized_head().await?))
+		Ok(Some(self.api.backend().latest_finalized_block_ref().await?.hash()))
 	}
 
 	/// Subscribe to new parachain blocks.
 	pub async fn on_block<F, R>(&self, on_block: F) -> Result<(), Error>
 	where
 		F: Fn(SpacewalkHeader) -> R,
 		R: Future<Output = Result<(), Error>>,
-	{
-		let mut sub = self.api.rpc().subscribe_finalized_block_headers().await?;
+	{	
+		let mut sub = self.legacy_rpc.chain_subscribe_finalized_heads().await?;
```

Comment:
If functions like this are considered legacy, is there a better practice now that we should rather use instead?

Reply:
It seems from the [CHANGELOG](https://github.com/paritytech/subxt/blob/master/CHANGELOG.md#0320---2023-09-27) that it is the "well maintained" one, while the other is `UnstableBackend`.